SingletonPattern/SingletonPattern.cpp

```diff
@@ -3,25 +3,30 @@ using namespace std;
 
 class Singleton {
 public:
-    static Singleton* Instance() {
-        if (!_instance) {
-            _instance = new Singleton();
-        }
-        return _instance;
+    static Singleton& Instance() {
+        static Singleton instance;
+        return instance;
     }
+
+    void printInstanceAddr() {
+        cout << this << endl;
+    }
+
 private:
-    Singleton() { cout << "Singleton..." << endl; }
-    static Singleton* _instance;
+    Singleton() {
+        cout << "ctor" << endl;
```

Comment:
What does `ctor` means here?

Reply:
It shows how many times the `ctor`  has been called.